File: inspectia_app/test_communication_complete.py
#!/usr/bin/env python3
"""
Test complet de communication Frontend-Backend pour InspectIA
Système ML-RL hybride version 2.0.0
"""

import requests
import json
import time
from datetime import datetime
from pathlib import Path

# Configuration
BASE_URL = "http://localhost:8000"
TEST_RESULTS = []

def log_test(test_name, success, details=""):
    """Logger les résultats de test"""
    status = "✅ PASSÉ" if success else "❌ ÉCHOUÉ"
    timestamp = datetime.now().strftime("%H:%M:%S")
    result = f"[{timestamp}] {status} {test_name}"
    if details:
        result += f" - {details}"
    print(result)
    TEST_RESULTS.append({
        "test": test_name,
        "success": success,
        "details": details,
        "timestamp": timestamp
    })

def test_backend_health():
    """Test 1: Santé du backend"""
    try:
        response = requests.get(f"{BASE_URL}/health", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if data.get("status") == "healthy":
                log_test("Santé Backend", True, f"Version {data.get('version')}")
                return True
        log_test("Santé Backend", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Santé Backend", False, str(e))
        return False

def test_chapters_endpoint():
    """Test 2: Endpoint des chapitres"""
    try:
        response = requests.get(f"{BASE_URL}/chapters", timeout=10)
        if response.status_code == 200:
            data = response.json()
            chapters = data.get("chapters", [])
            if len(chapters) >= 3:
                chapter_names = [c["name"] for c in chapters]
                log_test("Chapitres Disponibles", True, f"{len(chapters)} chapitres: {', '.join(chapter_names[:3])}")
                return True
        log_test("Chapitres Disponibles", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Chapitres Disponibles", False, str(e))
        return False

def test_chapter_config():
    """Test 3: Configuration d'un chapitre"""
    try:
        response = requests.get(f"{BASE_URL}/predict/chap30/config", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if data.get("chapter") == "chap30":
                log_test("Configuration Chapitre", True, "Chap30 configuré")
                return True
        log_test("Configuration Chapitre", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Configuration Chapitre", False, str(e))
        return False

def test_file_upload_prediction():
    """Test 4: Upload et prédiction de fichier"""
    try:
        # Créer un fichier CSV de test
        test_csv = "DECLARATION_ID,VALEUR_CAF,POIDS_NET_KG,NOMBRE_COLIS,QUANTITE_COMPLEMENT\nTEST_001,1000.0,10.5,1,0"
        
        files = {'file': ('test_declaration.csv', test_csv, 'text/csv')}
        data = {
            'declaration_id': 'TEST_001',
            'inspector_id': 'INSP_001'
        }
        
        response = requests.post(
            f"{BASE_URL}/predict/chap30", 
            files=files, 
            data=data,
            timeout=30
        )
        
        if response.status_code == 200:
            result = response.json()
            if "prediction" in result:
                pred = result["prediction"]
                log_test("Upload et Prédiction", True, f"Fraude: {pred.get('predicted_fraud', 'N/A')}")
                return True
        log_test("Upload et Prédiction", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Upload et Prédiction", False, str(e))
        return False

def test_rl_system():
    """Test 5: Système RL"""
    try:
        response = requests.get(f"{BASE_URL}/predict/chap30/rl/stats", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if "performance_summary" in data:
                stats = data["performance_summary"]
                total_decisions = stats.get("general_statistics", {}).get("total_decisions", 0)
                log_test("Système RL", True, f"{total_decisions} décisions enregistrées")
                return True
        log_test("Système RL", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Système RL", False, str(e))
        return False

def test_postgresql_connection():
    """Test 6: Connexion PostgreSQL"""
    try:
        response = requests.get(f"{BASE_URL}/api/v2/health/", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if data.get("database") == "postgresql" and data.get("connection") == "active":
                log_test("Base de Données PostgreSQL", True, "Connexion active")
                return True
        log_test("Base de Données PostgreSQL", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Base de Données PostgreSQL", False, str(e))
        return False

def test_ml_dashboard():
    """Test 7: Dashboard ML"""
    try:
        response = requests.get(f"{BASE_URL}/ml/ml-dashboard", timeout=15)
        if response.status_code == 200:
            data = response.json()
            if "data" in data and "performance" in data["data"]:
                performance = data["data"]["performance"]
                chapters = list(performance.keys())
                log_test("Dashboard ML", True, f"Données pour {len(chapters)} chapitres")
                return True
        log_test("Dashboard ML", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Dashboard ML", False, str(e))
        return False

def test_advanced_endpoints():
    """Test 8: Endpoints avancés"""
    try:
        # Test endpoint de détection de fraude
        response = requests.get(f"{BASE_URL}/predict/fraud-detection-methods", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if "data" in data and "advanced_techniques" in data["data"]:
                techniques = data["data"]["advanced_techniques"]
                log_test("Endpoints Avancés", True, f"{len(techniques)} techniques de détection")
                return True
        log_test("Endpoints Avancés", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Endpoints Avancés", False, str(e))
        return False

def test_system_status():
    """Test 9: Statut système"""
    try:
        response = requests.get(f"{BASE_URL}/predict/system-status", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if "overall_health" in data:
                health = data["overall_health"]
                chapters = len(data.get("chapters", {}))
                log_test("Statut Système", True, f"Santé: {health}, {chapters} chapitres")
                return True
        log_test("Statut Système", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Statut Système", False, str(e))
        return False

def test_dependencies():
    """Test 10: Dépendances"""
    try:
        response = requests.get(f"{BASE_URL}/predict/dependencies", timeout=10)
        if response.status_code == 200:
            data = response.json()
            if "dependencies" in data:
                deps = data["dependencies"]
                available = sum(1 for v in deps.values() if v)
                total = len(deps)
                log_test("Dépendances", True, f"{available}/{total} disponibles")
                return True
        log_test("Dépendances", False, f"Code: {response.status_code}")
        return False
    except Exception as e:
        log_test("Dépendances", False, str(e))
        return False

def main():
    """Fonction principale de test"""
    print("🧪 TEST COMPLET DE COMMUNICATION FRONTEND-BACKEND")
    print("=" * 60)
    print(f"🌐 URL Backend: {BASE_URL}")
    print(f"⏰ Début des tests: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print()
    
    # Liste des tests à exécuter
    tests = [
        ("Santé Backend", test_backend_health),
        ("Chapitres Disponibles", test_chapters_endpoint),
        ("Configuration Chapitre", test_chapter_config),
        ("Upload et Prédiction", test_file_upload_prediction),
        ("Système RL", test_rl_system),
        ("Base de Données PostgreSQL", test_postgresql_connection),
        ("Dashboard ML", test_ml_dashboard),
        ("Endpoints Avancés", test_advanced_endpoints),
        ("Statut Système", test_system_status),
        ("Dépendances", test_dependencies),
    ]
    
    # Exécution des tests
    passed = 0
    total = len(tests)
    
    for test_name, test_func in tests:
        try:
            if test_func():
                passed += 1
        except Exception as e:
            log_test(test_name, False, f"Erreur: {str(e)}")
        time.sleep(1)  # Pause entre les tests
    
    # Résumé
    print()
    print("📊 RÉSUMÉ DES TESTS")
    print("=" * 60)
    print(f"Total: {total}")
    print(f"Passés: {passed}")
    print(f"Échoués: {total - passed}")
    print(f"Taux de réussite: {(passed/total)*100:.1f}%")
    
    if passed == total:
        print("\n🎉 TOUS LES TESTS SONT PASSÉS!")
        print("✅ La communication Frontend-Backend fonctionne parfaitement")
    elif passed >= total * 0.8:
        print("\n⚠️ La plupart des tests sont passés")
        print("✅ La communication Frontend-Backend fonctionne globalement")
    else:
        print("\n❌ Plusieurs tests ont échoué")
        print("⚠️ Vérifiez la configuration du backend")
    
    # Sauvegarde des résultats
    results_file = Path("test_communication_results.json")
    with open(results_file, "w", encoding="utf-8") as f:
        json.dump({
            "timestamp": datetime.now().isoformat(),
            "summary": {
                "total": total,
                "passed": passed,
                "failed": total - passed,
                "success_rate": (passed/total)*100
            },
            "tests": TEST_RESULTS
        }, f, indent=2, ensure_ascii=False)
    
    print(f"\n📄 Résultats sauvegardés dans: {results_file}")

if __name__ == "__main__":
    main()

File: inspectia_app/test_communication_complete.dart
#!/usr/bin/env python3
"""
🧪 TESTS COMPLETS DE COMMUNICATION FRONTEND-BACKEND
Teste l'intégration complète entre le frontend Flutter et le backend Python
"""

import asyncio
import json
import time
import requests
import sys
from pathlib import Path
from typing import Dict, List, Any, Optional
import subprocess
import os

class FrontendBackendCommunicationTester:
    """Testeur complet de communication frontend-backend"""
    
    def __init__(self):
        self.base_url = "http://localhost:8000"
        self.frontend_path = Path("inspectia_app_frontend")
        self.backend_path = Path("backend")
        self.results = {
            "total_tests": 0,
            "passed_tests": 0,
            "failed_tests": 0,
            "test_details": []
        }
        
    def log_test(self, test_name: str, success: bool, details: str = "", error: str = ""):
        """Enregistre le résultat d'un test"""
        self.results["total_tests"] += 1
        if success:
            self.results["passed_tests"] += 1
            print(f"✅ {test_name}")
        else:
            self.results["failed_tests"] += 1
            print(f"❌ {test_name}: {error}")
            
        self.results["test_details"].append({
            "test_name": test_name,
            "success": success,
            "details": details,
            "error": error,
            "timestamp": time.time()
        })
    
    def test_backend_health(self) -> bool:
        """Test 1: Santé du backend"""
        try:
            response = requests.get(f"{self.base_url}/api/v1/health", timeout=10)
            if response.status_code == 200:
                data = response.json()
                if data.get("success") and data.get("data", {}).get("status") == "healthy":
                    self.log_test("Backend Health Check", True, f"Version: {data.get('data', {}).get('version')}")
                    return True
            self.log_test("Backend Health Check", False, "", f"Status: {response.status_code}")
            return False
        except Exception as e:
            self.log_test("Backend Health Check", False, "", str(e))
            return False
    
    def test_backend_endpoints(self) -> bool:
        """Test 2: Tous les endpoints backend"""
        endpoints_to_test = [
            ("/api/v1/chapters", "GET"),
            ("/api/v1/system/status", "GET"),
            ("/api/v1/config/chap30", "GET"),
            ("/api/v1/model/chap30/info", "GET"),
            ("/api/v1/features/chap30", "GET"),
            ("/api/v1/status/chap30", "GET"),
            ("/api/v1/ml/dashboard", "GET"),
            ("/api/v1/rl/performance/chap30/basic", "GET"),
            ("/api/v1/rl/analytics/chap30/basic", "GET"),
            ("/api/v1/retraining/chap30/status", "GET"),
            ("/api/v1/inspectors/chap30/profiles", "GET"),
            ("/api/v1/chef/dashboard", "GET"),
            ("/api/v1/declarations/", "GET"),
            ("/api/v1/postgresql/health", "GET"),
            ("/api/v1/postgresql/stats", "GET"),
        ]
        
        success_count = 0
        for endpoint, method in endpoints_to_test:
            try:
                if method == "GET":
                    response = requests.get(f"{self.base_url}{endpoint}", timeout=10)
                elif method == "POST":
                    response = requests.post(f"{self.base_url}{endpoint}", timeout=10)
                
                if response.status_code in [200, 201]:
                    success_count += 1
                else:
                    self.log_test(f"Endpoint {endpoint}", False, "", f"Status: {response.status_code}")
            except Exception as e:
                self.log_test(f"Endpoint {endpoint}", False, "", str(e))
        
        success_rate = success_count / len(endpoints_to_test)
        if success_rate >= 0.8:  # 80% de réussite minimum
            self.log_test("Backend Endpoints", True, f"{success_count}/{len(endpoints_to_test)} endpoints OK")
            return True
        else:
            self.log_test("Backend Endpoints", False, "", f"Only {success_count}/{len(endpoints_to_test)} endpoints OK")
            return False
    
    def test_prediction_endpoints(self) -> bool:
        """Test 3: Endpoints de prédiction"""
        test_data = {
            "declaration_id": "TEST_COMM_001",
            "valeur_caf": 15000.0,
            "poids_net_kg": 750.0,
            "nombre_colis": 15,
            "code_sh_complet": "30049000",
            "pays_origine": "FR",
            "bureau_douane": "01A"
        }
        
        chapters = ["chap30", "chap84", "chap85"]
        success_count = 0
        
        for chapter in chapters:
            try:
                # Test prédiction avec données
                response = requests.post(
                    f"{self.base_url}/api/v1/predict/{chapter}",
                    json=test_data,
                    timeout=15
                )
                
                if response.status_code == 200:
                    data = response.json()
                    if data.get("success") and "prediction" in data:
                        success_count += 1
                        self.log_test(f"Prediction {chapter}", True, f"Score: {data.get('prediction', {}).get('score', 'N/A')}")
                    else:
                        self.log_test(f"Prediction {chapter}", False, "", "Invalid response structure")
                else:
                    self.log_test(f"Prediction {chapter}", False, "", f"Status: {response.status_code}")
                    
            except Exception as e:
                self.log_test(f"Prediction {chapter}", False, "", str(e))
        
        success_rate = success_count / len(chapters)
        if success_rate >= 0.8:
            self.log_test("Prediction Endpoints", True, f"{success_count}/{len(chapters)} chapters OK")
            return True
        else:
            self.log_test("Prediction Endpoints", False, "", f"Only {success_count}/{len(chapters)} chapters OK")
            return False
    
    def test_database_connection(self) -> bool:
        """Test 4: Connexion à la base de données"""
        try:
            response = requests.get(f"{self.base_url}/api/v1/postgresql/health", timeout=10)
            if response.status_code == 200:
                data = response.json()
                if data.get("success") and data.get("data", {}).get("status") == "connected":
                    self.log_test("Database Connection", True, "PostgreSQL connected")
                    return True
            
            self.log_test("Database Connection", False, "", f"Status: {response.status_code}")
            return False
        except Exception as e:
            self.log_test("Database Connection", False, "", str(e))
            return False
    
    def test_file_upload(self) -> bool:
        """Test 5: Upload de fichiers"""
        try:
            # Créer un fichier de test
            test_file_content = """DECLARATION_ID,VALEUR_CAF,POIDS_NET_KG,NOMBRE_COLIS,CODE_SH_COMPLET,PAYS_ORIGINE,BUREAU_DOUANE
TEST_UPLOAD_001,10000.0,500.0,10,30049000,FR,01A
TEST_UPLOAD_002,8000.0,400.0,8,30049000,DE,01A"""
            
            test_file_path = Path("test_upload.csv")
            test_file_path.write_text(test_file_content)
            
            # Test upload
            with open(test_file_path, 'rb') as f:
                files = {'file': ('test_upload.csv', f, 'text/csv')}
                response = requests.post(
                    f"{self.base_url}/api/v1/upload/chap30",
                    files=files,
                    timeout=30
                )
            
            # Nettoyer le fichier de test
            test_file_path.unlink(missing_ok=True)
            
            if response.status_code in [200, 201]:
                data = response.json()
                if data.get("success"):
                    self.log_test("File Upload", True, f"Processed: {data.get('processed_count', 0)} files")
                    return True
            
            self.log_test("File Upload", False, "", f"Status: {response.status_code}")
            return False
            
        except Exception as e:
            self.log_test("File Upload", False, "", str(e))
            return False
    
    def test_rl_feedback_loop(self) -> bool:
        """Test 6: Boucle de feedback RL"""
        try:
            feedback_data = {
                "declaration_id": "TEST_RL_001",
                "predicted_fraud": False,
                "predicted_probability": 0.3,
                "inspector_decision": True,
                "inspector_confidence": 0.8,
                "inspector_id": "INSP001",
                "additional_notes": "Test feedback automatique",
                "exploration_used": True
            }
            
            chapters = ["chap30", "chap84", "chap85"]
            success_count = 0
            
            for chapter in chapters:
                response = requests.post(
                    f"{self.base_url}/api/v1/rl/feedback/{chapter}",
                    json=feedback_data,
                    timeout=10
                )
                
                if response.status_code == 200:
                    data = response.json()
                    if data.get("success"):
                        success_count += 1
                        self.log_test(f"RL Feedback {chapter}", True, "Feedback submitted")
                    else:
                        self.log_test(f"RL Feedback {chapter}", False, "", "Invalid response")
                else:
                    self.log_test(f"RL Feedback {chapter}", False, "", f"Status: {response.status_code}")
            
            success_rate = success_count / len(chapters)
            if success_rate >= 0.8:
                self.log_test("RL Feedback Loop", True, f"{success_count}/{len(chapters)} chapters OK")
                return True
            else:
                self.log_test("RL Feedback Loop", False, "", f"Only {success_count}/{len(chapters)} chapters OK")
                return False
                
        except Exception as e:
            self.log_test("RL Feedback Loop", False, "", str(e))
            return False
    
    def test_frontend_flutter_tests(self) -> bool:
        """Test 7: Tests Flutter frontend"""
        try:
            # Vérifier si Flutter est installé
            flutter_check = subprocess.run(["flutter", "--version"], capture_output=True, text=True)
            if flutter_check.returncode != 0:
                self.log_test("Flutter Tests", False, "", "Flutter not installed")
                return False
            
            # Changer vers le répertoire frontend
            frontend_dir = Path("inspectia_app/inspectia_app_frontend")
            if not frontend_dir.exists():
                self.log_test("Flutter Tests", False, "", "Frontend directory not found")
                return False
            
            # Exécuter les tests Flutter
            os.chdir(frontend_dir)
            test_result = subprocess.run(
                ["flutter", "test", "test_frontend_backend.dart"],
                capture_output=True,
                text=True,
                timeout=120
            )
            
            # Retourner au répertoire principal
            os.chdir("../..")
            
            if test_result.returncode == 0:
                self.log_test("Flutter Tests", True, "All Flutter tests passed")
                return True
            else:
                self.log_test("Flutter Tests", False, "", f"Flutter test errors: {test_result.stderr}")
                return False
                
        except Exception as e:
            self.log_test("Flutter Tests", False, "", str(e))
            return False
    
    def test_performance_benchmarks(self) -> bool:
        """Test 8: Benchmarks de performance"""
        try:
            # Test de latence
            start_time = time.time()
            response = requests.get(f"{self.base_url}/api/v1/health", timeout=10)
            end_time = time.time()
            
            latency = end_time - start_time
            
            # Test de débit (multiple requêtes)
            start_time = time.time()
            for _ in range(10):
                requests.get(f"{self.base_url}/api/v1/chapters", timeout=5)
            end_time = time.time()
            
            throughput_time = end_time - start_time
            throughput = 10 / throughput_time  # requêtes par seconde
            
            # Critères de performance
            latency_ok = latency < 2.0  # moins de 2 secondes
            throughput_ok = throughput > 5.0  # plus de 5 req/s
            
            if latency_ok and throughput_ok:
                self.log_test("Performance Benchmarks", True, 
                            f"Latency: {latency:.2f}s, Throughput: {throughput:.1f} req/s")
                return True
            else:
                self.log_test("Performance Benchmarks", False, "", 
                            f"Latency: {latency:.2f}s (>2s), Throughput: {throughput:.1f} req/s (<5 req/s)")
                return False
                
        except Exception as e:
            self.log_test("Performance Benchmarks", False, "", str(e))
            return False
    
    def test_error_handling(self) -> bool:
        """Test 9: Gestion d'erreurs"""
        try:
            # Test endpoint inexistant
            response = requests.get(f"{self.base_url}/api/v1/nonexistent", timeout=10)
            if response.status_code == 404:
                error_handling_ok = True
            else:
                error_handling_ok = False
            
            # Test données invalides
            invalid_data = {"invalid": "data"}
            response = requests.post(f"{self.base_url}/api/v1/predict/chap30", 
                                   json=invalid_data, timeout=10)
            if response.status_code in [400, 422]:
                error_handling_ok = error_handling_ok and True
            else:
                error_handling_ok = False
            
            if error_handling_ok:
                self.log_test("Error Handling", True, "Proper error responses")
                return True
            else:
                self.log_test("Error Handling", False, "", "Incorrect error responses")
                return False
                
        except Exception as e:
            self.log_test("Error Handling", False, "", str(e))
            return False
    
    def test_data_consistency(self) -> bool:
        """Test 10: Cohérence des données"""
        try:
            # Test cohérence entre différents endpoints
            chapters_response = requests.get(f"{self.base_url}/api/v1/chapters", timeout=10)
            health_response = requests.get(f"{self.base_url}/api/v1/health", timeout=10)
            
            if chapters_response.status_code == 200 and health_response.status_code == 200:
                chapters_data = chapters_response.json()
                health_data = health_response.json()
                
                # Vérifier la cohérence
                if (chapters_data.get("success") and 
                    health_data.get("success") and 
                    len(chapters_data.get("chapters", [])) > 0):
                    
                    self.log_test("Data Consistency", True, 
                                f"Chapters: {len(chapters_data.get('chapters', []))}, Health: OK")
                    return True
            
            self.log_test("Data Consistency", False, "", "Inconsistent data between endpoints")
            return False
            
        except Exception as e:
            self.log_test("Data Consistency", False, "", str(e))
            return False
    
    def run_all_tests(self) -> Dict[str, Any]:
        """Exécute tous les tests"""
        print("🚀 DÉMARRAGE DES TESTS COMPLETS DE COMMUNICATION FRONTEND-BACKEND")
        print("=" * 80)
        
        # Liste des tests à exécuter
        tests = [
            ("Backend Health", self.test_backend_health),
            ("Backend Endpoints", self.test_backend_endpoints),
            ("Prediction Endpoints", self.test_prediction_endpoints),
            ("Database Connection", self.test_database_connection),
            ("File Upload", self.test_file_upload),
            ("RL Feedback Loop", self.test_rl_feedback_loop),
            ("Flutter Tests", self.test_frontend_flutter_tests),
            ("Performance Benchmarks", self.test_performance_benchmarks),
            ("Error Handling", self.test_error_handling),
            ("Data Consistency", self.test_data_consistency),
        ]
        
        # Exécution des tests
        for test_name, test_func in tests:
            print(f"\n🧪 Test: {test_name}")
            try:
                test_func()
            except Exception as e:
                self.log_test(test_name, False, "", f"Test crashed: {str(e)}")
            time.sleep(1)  # Pause entre les tests
        
        # Résumé final
        self.print_summary()
        return self.results
    
    def print_summary(self):
        """Affiche le résumé des tests"""
        print("\n" + "=" * 80)
        print("📊 RÉSUMÉ DES TESTS DE COMMUNICATION FRONTEND-BACKEND")
        print("=" * 80)
        
        total = self.results["total_tests"]
        passed = self.results["passed_tests"]
        failed = self.results["failed_tests"]
        success_rate = (passed / total * 100) if total > 0 else 0
        
        print(f"📈 Total des tests: {total}")
        print(f"✅ Tests réussis: {passed}")
        print(f"❌ Tests échoués: {failed}")
        print(f"📊 Taux de réussite: {success_rate:.1f}%")
        
        if success_rate >= 90:
            print("🎉 EXCELLENT! Communication frontend-backend très stable")
        elif success_rate >= 80:
            print("👍 BIEN! Communication frontend-backend stable")
        elif success_rate >= 70:
            print("⚠️ ACCEPTABLE! Quelques problèmes mineurs détectés")
        else:
            print("🚨 PROBLÈME! Communication frontend-backend défaillante")
        
        # Détails des échecs
        if failed > 0:
            print("\n❌ TESTS ÉCHOUÉS:")
            for test in self.results["test_details"]:
                if not test["success"]:
                    print(f"  - {test['test_name']}: {test['error']}")
        
        print("\n" + "=" * 80)

def main():
    """Fonction principale"""
    tester = FrontendBackendCommunicationTester()
    
    print("🔧 Vérification de l'environnement...")
    
    # Vérifier si le backend est démarré
    try:
        response = requests.get("http://localhost:8000/api/v1/health", timeout=5)
        if response.status_code != 200:
            print("⚠️ Backend non démarré ou non accessible sur http://localhost:8000")
            print("💡 Démarrez le backend avec: cd backend && python -m uvicorn src.main:app --reload")
            return
    except:
        print("⚠️ Backend non démarré ou non accessible sur http://localhost:8000")
        print("💡 Démarrez le backend avec: cd backend && python -m uvicorn src.main:app --reload")
        return
    
    print("✅ Backend accessible, démarrage des tests...")
    
    # Exécuter tous les tests
    results = tester.run_all_tests()
    
    # Sauvegarder les résultats
    results_file = Path("test_communication_results.json")
    with open(results_file, 'w') as f:
        json.dump(results, f, indent=2)
    
    print(f"📄 Résultats sauvegardés dans: {results_file}")

if __name__ == "__main__":
    main()
